### DISCUSSION TITLE
Serving dev server on LAN

### DISCUSSION THREAD

**Discussion by tigawanna** [pocketbase/pocketbase#435] [Q&A] [answered]
is there a native way to accomplish this? I use this when testing across devices

**Comment by Deniswarui4** [comment]
Kinuthia Dennis, here is another Denis warui. Let's link up on WhatsApp:
±254722593572

**Comment by ganigeorgiev** [comment, marked as answer]
There are several ways to expose your local instance, depending on how is your network and firewall configured.

I usually use something like [ngrok](https://ngrok.com/) when I want to test or access my localhost from an external device.

Another option could be also to bind to the IP of your local network (or directly to `0.0.0.0`), eg.:
```
./pocketbase serve --http="192.168.0.1:8090"
```

**Comment by tigawanna** [comment]
## am running into this issue

```
2022/09/14 14:38:22 listen tcp 192.168.43.238:8090: bind: cannot assign requested address
```

*detailed error*

```sh
⋊> ~/pocketbase ./pocketbase serve --debug --http="192.168.43.238:8090"                                          14:38:10
[0.00ms] SELECT {{_params}}.* FROM `_params` WHERE `key`='settings' LIMIT 1
[0.00ms] CREATE TABLE IF NOT EXISTS `_migrations` (file VARCHAR(255) PRIMARY KEY NOT NULL, applied INTEGER NOT NULL)
[0.00ms] SELECT count(*) FROM `_migrations` WHERE `file`='1640988000_init.go' LIMIT 1
[0.00ms] SELECT count(*) FROM `_migrations` WHERE `file`='1661586591_add_externalAuths_table.go' LIMIT 1
[0.00ms] SELECT {{_params}}.* FROM `_params` WHERE `key`='settings' LIMIT 1
> Server started at: http://192.168.43.238:8090
  - REST API: http://192.168.43.238:8090/api/
  - Admin UI: http://192.168.43.238:8090/_/
2022/09/14 14:38:22 listen tcp 192.168.43.238:8090: bind: cannot assign requested address
```

**Reply by ganigeorgiev** [reply]
It seems that the provided IP is not your local address, try running something like `ifconfig | grep netmask` to ensure that the correct local IP is used.

Another reason why it may fail is that there is maybe a firewall (eg. selinux) that blocks outcoming/incoming localhost network requests or binding to the specified port.

You can also try binding to the designated `0.0.0.0` IP, which basically means "bind to all available IPv4 on the local machine".

---

But [ngrok](https://ngrok.com/) is also a very good and easy to setup alternative (it has free plan), so I highly recommend it. Another alternative to ngrok that was mentioned here in the past is https://localxpose.io/.

**Reply by tigawanna** [reply]
thank you, it worked, I was on WSL ubuntu and couldn't figure that one out 
i tried 
```
⋊> ~ ifconfig | grep netmask                                                                                     15:00:39
        inet 172.30.129.16  netmask 255.255.240.0  broadcast 172.30.143.255
        inet 127.0.0.1  netmask 255.0.0.0
```
and 
```sh
⋊> ~/pocketbase cat /etc/resolv.conf                                                                             15:04:04
# This file was automatically generated by WSL. To stop automatic generation of this file, add the following entry to /etc/wsl.conf:
# [network]
# generateResolvConf = false
nameserver 172.30.128.1

```
but the windows version works
ipconfig gave me ipv4 192.168.43.238
and ran this with it 
```sh
 .\pocketbase.exe serve --http="192.168.43.238:8090"
```
